### DISCUSSION TITLE
attributeValue restriction

### DISCUSSION THREAD

**Discussion by sschulz-t** [openwallet-foundation/credo-ts#1277] [Q&A] [answered]
I am trying to use an attributeValue restriction in my own code on top of aries-framework-javascript in combination with the BC wallet app (https://www2.gov.bc.ca/gov/content/governments/government-id/bc-wallet).

My proof request is using the following attributes in the request in order to make sure the first name is David:
```
...
"requestedAttributes" : {
    "first_name": {
      "name": "first_name",
      "restrictions": [
        { "attributeValue": 
          { 
            "name": "first_name", 
            "value": "David"
          }
        },
       { "credentialDefinitionId": "xxxxxx"}
      ]
    }
  },
...
```

However this does not seem to work, the wallet app complains that i do not have a matching credential (which i do have ;) )

I found this issue: https://github.com/hyperledger/aries-framework-javascript/issues/1033 
Is this the problem I am facing? 

At which point should the AttributeFilterTransformer kick in? During proof generation in the wallet app?

**Comment by TimoGlastra** [comment, marked as answer]
Hi @sschulz-t, I think you're indeed running into the issue as described in #1033. We've completely overhauled the anoncreds implementation and it should be fixed in the next big release (0.4.0)
